### ISSUE TITLE
The first time the app is opened and the map is displayed, touch on back key closes the app (Android)

### ISSUE THREAD

**Issue by andresigualada** [phonegap-googlemaps-plugin/cordova-plugin-googlemaps#674]
Related to this issue: https://github.com/phonegap-googlemaps-plugin/cordova-plugin-googlemaps/issues/5

When the app is opened for the first time and the map is displayed, touching the back key will close the app in some android devices (Tested on One Plus One). However, this isn't happening the following times the app is opened.

This is not related to markers or infowindows. I've tested it just creating the map without markers and events.

**Comment by Hirbod** [comment]
I'm sorry, not able to reproduce this. You should listen to the back button event and prevent from closing. Ionic and OnsenUI do have defaults on board which you can use.

OnsenUI Example
```js

        ons.setDefaultDeviceBackButtonListener(function() {
            if (navigator.notification.confirm("Do you really want to close the App?", 
                function(index) {
                    if (index == 1) { // OK button
                        navigator.app.exitApp(); // Close the app
                    }
                }
            ));
        });

```

**Comment by andresigualada** [comment]
Thanks for your quickly response, Hirbod. Anyway, I was already capturing the "backbutton" event and the bug still happens. My implementation it's the same as in cordova docs: http://cordova.apache.org/docs/en/edge/cordova/events/events.backbutton.html

If you have any other idea about this, I will appreciate it. Thanks!

**Comment by Hirbod** [comment]
Are you on crosswalk?

**Comment by andresigualada** [comment]
Nope, I'm using "plain" cordova with some other plugins. To debug I use "adb logcat" on the device and chrome inspector.

I think it could be a device-related issue, because I've tested it and happens in Galaxy 4, 5, One Plus One. However, all the tests I've done in Galaxy Mini 3 works fine.

"adb logcat" doesn't throw any error or info related

**Comment by Hirbod** [comment]
This is an acitvity problem, and I don't have enough know how to fix this.
You should consider installing crosswalk. (always, it just eliminates a lot of problems and you will have a consistent webview across all platforms)

It will just take 5 minutes of work. Just try it, and please tell me if it fixed your problems.
https://github.com/phonegap-googlemaps-plugin/cordova-plugin-googlemaps/wiki/Tutorial-for-CrossWalk-Webview-Plugin-%28Android%29

**Comment by andresigualada** [comment]
Thank you for your support. I will try crosswalk and let you know if I find anything.

**Comment by Hirbod** [comment]
Any news?

**Comment by andresigualada** [comment]
Nope, I was playing with crosswalk with no luck. I have been watching the logs but found nothing..

**Comment by Hirbod** [comment]
Newest Version? minSdkVersion 16 ?

**Comment by andresigualada** [comment]
I'm using Cordova 5.3.3 (latest I think) and minSdkVersion is 15.

It's really strange, because only happens the first time the app is opened. These are the latest lines of the logcat before the app crash, maybe it's useful to you:

```
D/WebView ( 6414): loadUrlImpl: called
D/webcore ( 6414):  CORE loadUrl: called
D/webkit  ( 6414): Firewall not null
D/webkit  ( 6414): euler: isUrlBlocked = false
I/dalvikvm( 6414): Jit: resizing JitTable from 8192 to 16384
E/SQLiteLog( 6414): (14) cannot open file at line 32516 of [00bb9c9ce4]
E/SQLiteLog( 6414): (14) os_unix.c:32516: (2) open(/NotificationPermissions.db) - 
D/WebKit  ( 6414): ERROR: 
D/WebKit  ( 6414): SQLite database failed to load from /NotificationPermissions.db
D/WebKit  ( 6414): Cause - unable to open database file
D/WebKit  ( 6414): 
D/WebKit  ( 6414): external/webkit/Source/WebCore/platform/sql/SQLiteDatabase.cpp(71) : bool WebCore::SQLiteDatabase::open(const WTF::String&, bool)
D/AbsListView( 6414): unregisterIRListener() is called 
W/IInputConnectionWrapper( 6414): showStatusIcon on inactive InputConnection
D/WebView ( 6414): loadUrlImpl: called
D/webcore ( 6414):  CORE loadUrl: called
D/webkit  ( 6414): Firewall not null
D/webkit  ( 6414): euler: isUrlBlocked = false
W/sqlite4java( 6414): [sqlite] DB[1]: will not dispose from a non-confining thread
D/WebView ( 6414): loadUrlImpl: called
I/GATE    ( 6414): <GATE-M>DEV_ACTION_COMPLETED</GATE-M>
D/webcore ( 6414):  CORE loadUrl: called
D/webkit  ( 6414): Firewall not null
D/webkit  ( 6414): euler: isUrlBlocked = false
E/webview ( 6414): java.lang.Throwable: Error: WebView.destroy() called while still attached!
E/webview ( 6414): 	at android.webkit.WebViewClassic.destroy(WebViewClassic.java:4158)
E/webview ( 6414): 	at android.webkit.WebView.destroy(WebView.java:707)
E/webview ( 6414): 	at org.apache.cordova.engine.SystemWebViewEngine.destroy(SystemWebViewEngine.java:324)
E/webview ( 6414): 	at org.apache.cordova.CordovaWebViewImpl.handleDestroy(CordovaWebViewImpl.java:483)
E/webview ( 6414): 	at org.apache.cordova.CordovaActivity.onDestroy(CordovaActivity.java:311)
E/webview ( 6414): 	at android.app.Activity.performDestroy(Activity.java:5543)
E/webview ( 6414): 	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1134)
E/webview ( 6414): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:3619)
E/webview ( 6414): 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:3654)
E/webview ( 6414): 	at android.app.ActivityThread.access$1300(ActivityThread.java:159)
E/webview ( 6414): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1369)
E/webview ( 6414): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/webview ( 6414): 	at android.os.Looper.loop(Looper.java:176)
E/webview ( 6414): 	at android.app.ActivityThread.main(ActivityThread.java:5419)
E/webview ( 6414): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/webview ( 6414): 	at java.lang.reflect.Method.invoke(Method.java:525)
E/webview ( 6414): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1046)
E/webview ( 6414): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:862)
E/webview ( 6414): 	at dalvik.system.NativeStart.main(Native Method)
D/webviewglue( 6414): nativeDestroy view: 0x59286d70
D/AbsListView( 6414): onDetachedFromWindow
I/Choreographer( 6414): Skipped 33 frames!  The application may be doing too much work on its main thread.
I/GATE    ( 6414): <GATE-M>DEV_ACTION_COMPLETED</GATE-M>
D/webviewglue( 6414): nativeDestroy view: 0x5bcefc30
I/ActivityManager( 2393): Process com.comerleads.infoisinfo (pid 6414) (adj 15) has died.
```

**Comment by Hirbod** [comment]
Could you try minSdkVersion 16 ?

> cordova run android -- --minSdkVersion=16

Please use it exactly like that (with double -- space --minSdkVersion=16)
This will not change your settings but will compile one-time with API-Level 16

**Comment by andresigualada** [comment]
No luck... still happening. Do you have any clue? So I could have a starting point and investigate deeper

**Comment by Hirbod** [comment]
You have different errors in you log. Did you try a naked project? Only with the maps plugin?

**Comment by andresigualada** [comment]
I have tried what you said and everything seems to work fine.

I found the cause of the problem, which is not related to googlemaps plugin. The error happens when using admob plugin (admob-google-cordova) and this one, so I would like to apologise for the time and inconveniences.

I haven't found the fix yet but in case anyone else bump into this, check the admob plugin because are conflicting in some way.
